### DISCUSSION TITLE
How to securely pass data from tenant to tenant?

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#10491] [Q&A]
We have authentication data coming to the default tenant, then it must pass that data to the particular client's tenant in order to sign a user in.
So I need a mechanism similar to `TempData` to pass the data from one tenant to another tenant without disclosing it via URL.

**Comment by hishamco** [comment]
Are you working on something similar to SSO?

**Reply by xperiandri** [reply]
Something like that. The system that I integrate with open a default route and passes tokens and tenant name via POST request.
So that I must manually redirect user to the correct tenant and manually sign him in.

**Reply by hishamco** [reply]
I already start sort of impersonation long time ago, after awhile I notice that Lombiq did something similar with tenants, so the use can access another tenant with thier identity, perhaps this may help you. @Piedone it would be useful if you could refer to what you did. may be it will a give a clue to @xperiandri

**Reply by Piedone** [reply]
Thanks for the shoutout Hisham, yes, this is it: https://github.com/Lombiq/Hosting-Tenants/blob/dev/Lombiq.Hosting.Tenants.Admin.Login/Readme.md A feature to let you log in from the Default tenant to any other tenant with their admin users.

**Comment by Piedone** [comment]
You can always pass data to a tenant via HTTP in the same way you can pass data to any other web app (like POST), however, you needn't go through the request for that: You can also access services across tenants, i.e. call code running in the context of another tenant, and do data sharing like that. See here: https://orcharddojo.net/blog/how-to-access-services-from-another-tenant-in-orchard-core-orchard-nuggets
